src/main/scala/is/hail/methods/LinearMixedRegression.scala

```diff
@@ -77,279 +77,179 @@ object LinearMixedRegression {
       if (delta <= 0d)
         fatal(s"delta must be positive, got ${ delta }"))
 
-    val covNames = "intercept" +: covExpr
-
     val n = y.length
-    val k = cov.cols
-    val d = n - k - 1
+    val c = C.cols
+    val d = n - c - 1
 
     if (d < 1)
-      fatal(s"lmmreg: $n samples and $k ${plural(k, "covariate")} including intercept implies $d degrees of freedom.")
+      fatal(s"lmmreg: $n samples and $c ${ plural(c, "covariate") } including intercept implies $d degrees of freedom.")
 
-    info(s"lmmreg: running lmmreg on $n samples with $k sample ${plural(k, "covariate")} including intercept...")
+    info(s"lmmreg: Running lmmreg on $n samples with $c sample ${ plural(c, "covariate") } including intercept...")
 
     optDelta match {
       case Some(del) => info(s"lmmreg: Delta of $del specified by user")
       case None => info(s"lmmreg: Estimating delta using ${ if (useML) "ML" else "REML" }... ")
     }
 
-    val Eigendecomposition(_, rowIds, evects, evals) = eigenDecomposition.filterRows(vds.sSignature, completeSamplesSet)
-    
-    if (! completeSamples.sameElements(rowIds))
+    val Eigendecomposition(_, rowIds, evects, evals) = eigen.filterRows(vds.sSignature, completeSamplesSet)
+
+    if (!completeSamples.sameElements(rowIds))
       fatal("Bad stuff")
```

Comment:
We should be more specific here so that if a bug causes this to happen we get a message about what actually went wrong.